### DISCUSSION TITLE
[Idea] Improve VBS deployment with a check is installed correctly or retry

### DISCUSSION THREAD

**Discussion by kintaro1981** [glpi-project/glpi-agent#505] [Ideas]
### Your idea

Hello,

I noticed that if I run the VBS and meanwhile another MSI is running (usually quietly in background) GLPI-Agent's msi obviously don't get installed as Windows can run one MSI at a time. If this happens the VBS doesn't notice the failed installation and fusioninventory get uninstalled.

This produce an issue, if you don't double check or if it's the vbs is launched remotely the result is that client is without any agent. (fusion uninstalled and glpi-agent not installed).

How about add a check if the msi installation is successfull?
Maybe running again the IsInstallationNeeded function before the end to do a check?

**Comment by g-bougard** [comment]
Hi @kintaro1981 

thank you for sharing this problem. I've to try to reproduce it by running another MSI install to see what error we can catch.

**Comment by g-bougard** [comment]
Hi @kintaro1981 

I was able to reproduce the problem and I agree this is still an important issue.

So I also included an update in the PR #507 for this request.
You can try it by downloading it directly here: https://raw.githubusercontent.com/glpi-project/glpi-agent/9fbc65fce120a9fca84d2912e87e4fa010ec25e0/contrib/windows/glpi-agent-deployment.vbs
I put some explanation in the PR description.

After you tried it and can you comment the PR and/or this issue ?

Thank you

**Reply by kintaro1981** [reply]
I confirm are working both local installation and MsiServer availability check.
Thank you

**Comment by kintaro1981** [comment]
I'm going to try it in the next days. In case of error installing GLPI-Agent the vbs is going to avoid the uninstallation of fusioninventory/ocs? Reading the main section I understand that the uninstallation happens before the installation of GLPI, right?

**Reply by g-bougard** [reply]
Hi @kintaro1981 

indeed, on this point, I did nothing because we have to uninstall older agent before to avoid glpi-agent trying to use a still opened port.

Maybe we should first just try to stop the currently installed agent service so the httpd port is released. And then only uninstall the older agent if glpi-agent installation is successful... but to be honest, I think we should not trust older agents to always stop its service in time. You still can implement yourself this algorithm: from a simple script, stop first the older agent service, start the glpi-agent installation, and maybe you can abort the installation if the older agent stop failed. Then on successful installation, uninstall the older agent or restart the service.

Anyway, now, I'm far more confident this is not a big deal as the script should not fail because of another MSI package installation. If the installation fails now, this would be for a more important reason that would probably require some human administrator investigations. Also in case of failure, the script can now report a useful returncode if Verbose option is set to yes.
